src/raster/rasterband.rs

```diff
@@ -863,6 +954,50 @@ pub struct StatisticsAll {
     pub std_dev: f64,
 }
 
+#[derive(Debug)]
+pub struct Histogram {
+    min: f64,
+    max: f64,
+    n_buckets: i32,
+    histogram: *mut i32,
+    histogram_vec: Option<Vec<i32>>,
+}
+
+impl Histogram {
+    /// Histogram lower bound
+    pub fn min(&self) -> f64 {
+        self.min
+    }
+
+    /// Histogram upper bound
+    pub fn max(&self) -> f64 {
+        self.max
+    }
+    /// Histogram values for each bucket
+    pub fn histogram(&self) -> &[i32] {
+        if let Some(hist) = self.histogram_vec.as_ref() {
+            return hist.as_slice();
+        }
+
+        unsafe { std::slice::from_raw_parts(self.histogram, self.n_buckets as usize) }
+    }
+
+    /// Histogram bucket size
+    pub fn bucket_size(&self) -> f64 {
+        (self.max - self.min) / self.histogram().len() as f64
+    }
+}
+
+impl Drop for Histogram {
+    fn drop(&mut self) {
+        if self.histogram_vec.is_none() {
+            unsafe {
+                gdal_sys::VSIFree(self.histogram as *mut libc::c_void);
```

Comment:
If it turns out that double storage mechanisms are indeed necessary, please document the reason and any associated assumptions. I'd personally go further and create an `enum` capturing the two storage cases to enforce the logic with types, e.g.:

```
enum HistCounts {
   UnsafePointer(*mut libc::c_void),
   OwnedVec(Vec<i32>)
}
```

But I think unifying it is preferable.